File: app/src/main/java/com/kobrakid/retroachievements/viewpager/ToggleableViewPager.java
package com.kobrakid.retroachievements.viewpager;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.ViewPager;

/**
 * Custom implementation of the ViewPager widget, allows for toggling paging.
 */
public class ToggleableViewPager extends ViewPager {

    private boolean pagingEnabled = true;

    public ToggleableViewPager(@NonNull Context context) {
        super(context);
    }

    public ToggleableViewPager(@NonNull Context context, AttributeSet attrs) { super(context, attrs); }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) { return pagingEnabled && super.onInterceptTouchEvent(ev); }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return pagingEnabled && super.onTouchEvent(ev);
    }

    public void setPagingEnabled(boolean pagingEnabled) {
        this.pagingEnabled = pagingEnabled;
    }

}

File: app/src/main/java/com/kobrakid/retroachievements/view/viewpager/ToggleableViewPager.kt
package com.kobrakid.retroachievements.view.viewpager

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

/**
 * Custom implementation of the ViewPager widget, allows for toggling paging.
 */
class ToggleableViewPager : ViewPager {

    private var pagingEnabled = true

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return pagingEnabled && super.onInterceptTouchEvent(ev)
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return pagingEnabled && super.onTouchEvent(ev)
    }

    fun setPagingEnabled(pagingEnabled: Boolean) {
        this.pagingEnabled = pagingEnabled
    }
}